modules/parquet/src/lib/parsers/parse-parquet-wasm.ts

```diff
@@ -3,32 +3,70 @@
 // Copyright (c) vis.gl contributors
 
 // eslint-disable
-import type {ArrowTable} from '@loaders.gl/arrow';
+import type {ReadableFile} from '@loaders.gl/loader-utils';
+import type {ArrowTable, ArrowTableBatch, Schema} from '@loaders.gl/schema';
 import {serializeArrowSchema} from '@loaders.gl/arrow';
-import type {ParquetWasmLoaderOptions} from '../../parquet-wasm-loader';
-import {loadWasm} from '../utils/load-wasm';
+
+import type * as parquetWasm from 'parquet-wasm';
 import * as arrow from 'apache-arrow';
 
-export async function parseParquetWasm(
-  arrayBuffer: ArrayBuffer,
-  options: ParquetWasmLoaderOptions
+import {loadWasm} from '../utils/load-wasm';
+import {makeStreamIterator} from '../utils/make-stream-iterator';
+
+export async function parseParquetFileWasm(
+  file: ReadableFile,
+  options?: parquetWasm.ReaderOptions & {wasmUrl: string}
 ): Promise<ArrowTable> {
-  const arr = new Uint8Array(arrayBuffer);
+  const wasmUrl = options?.wasmUrl;
+
+  const wasm = await loadWasm(wasmUrl);
+
+  let parquetFile: parquetWasm.ParquetFile;
+  if (file.handle instanceof Blob) {
+    // TODO - let's assume fromFile() works on Blobs and not just on File...
```

Comment:
Yes, I'll publish a patch with this fix: https://github.com/kylebarron/parquet-wasm/pull/530

Reply:
Published 0.6.1 where this uses the Blob type: https://kylebarron.dev/parquet-wasm/classes/esm_parquet_wasm.ParquetFile.html#fromFile